packages/headless/src/features/result-actions/result-actions-insight-analytics-actions.ts

```diff
@@ -21,13 +22,38 @@ export const logCopyToClipboard = (result: Result): InsightAction =>
         documentIdentifier(result),
         metadata
       );
-    }
-  );
+    },
+    analyticsType: 'InsightPanel.ItemAction',
+    analyticsPayloadBuilder: (state): InsightPanel.ItemAction => {
+      const metadata = getCaseContextAnalyticsMetadata(
+        state.insightCaseContext
+      );
+      const identifier = documentIdentifier(result);
+      const information = partialDocumentInformation(result, state);
+      return {
+        itemMetadata: {
+          uniqueFieldName: identifier.contentIDKey,
+          uniqueFieldValue: identifier.contentIDValue,
+          title: information.documentTitle,
+          author: information.documentAuthor,
+          url: information.documentUri,
+        },
+        position: information.documentPosition,
+        searchUid: state.search?.response.searchUid || '',
+        action: 'copyToClipboard',
+        context: {
+          targetId: metadata.caseId || '',
+          targetType: 'Case',
+          caseNumber: metadata.caseNumber,
+        } as InsightPanel.Context,
+      };
```

Comment:
I think everything here could be placed into some function to generate 
`itemMetadata, position, searchUid` <-- in 1 method
`context` <-- in another method
And we would only be left setting the `action` in the specific InsightPanel.ItemAction.

What do you think?
I wouldn't mind having a `features/analytics/insight-analytics-utils.ts` file with some generic functions in it.

You have this code in

`result-insight-analytics-actions`
`result-actions-insight-analytics-actions`
`attached-results-analytics-actions.ts`